src/m365/aad/commands/policy/policy-list.ts

```diff
@@ -0,0 +1,181 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import request from '../../../../request';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  policyType?: string;
+}
+
+class AadPolicyListCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return commands.POLICY_LIST;
+  }
+
+  public get description(): string {
+    return 'Returns policies from Azure AD';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.policyType = args.options.policyType || 'All';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const policyType: string = args.options.policyType ? args.options.policyType : 'all';
+    if (policyType && policyType !== "all") {
+      let endpoint = this.getPolicyEndPoint[policyType];
+      const url: string = `${this.resource}/v1.0/policies/${endpoint}`;
+      if (endpoint === "authorizationPolicy" || endpoint === "identitySecurityDefaultsEnforcementPolicy") {
+        ((): Promise<any> => {
+          return this.getPolicy(url);
+        })()
+          .then((policy: any): void => {
+            if (policy) {
+              logger.log(policy);
+            }
+            cb();
```

Comment:
Could you please brief on this?

Reply:
In multiple places in the code we have pieces of code similar to the above: output the retrieved policy and finish the command (`cb()`). To avoid repetition, we should see the command's code responsibilities as two actions: retrieve the policy and send the result to output and close the command. While we have multiple ways to retrieve the policy, once it's retrieved, sending it to output and closing the command is the same, so we shouldn't repeating that code. Does that help?

Reply:
> In multiple places in the code we have pieces of code similar to the above: output the retrieved policy and finish the command (`cb()`). To avoid repetition, we should see the command's code responsibilities as two actions: retrieve the policy and send the result to output and close the command. While we have multiple ways to retrieve the policy, once it's retrieved, sending it to output and closing the command is the same, so we shouldn't repeating that code. Does that help?

@waldekmastykarz, Thank you so much for your time to review ! I have simiplfied the code and fixed all the feedback points as per your suggestions. Could you please review and let me know if we need to improve anything in the code?